qiskit/circuit/register.py

```diff
@@ -82,7 +82,7 @@ def __init__(self, size=None, name=None, bits=None):
             )
         size = int(size)  # cast to int
 
-        if size <= 0:
+        if size < 0:
             raise CircuitError(
                 "Register size must be positive (%s '%s' was provided)"
```

Comment:
the error should now say "non-negative"

Reply:
@ajavadia As a mathematician positive includes zero (and strictly positive does not include zero). But the term non-negative is clearer indeed.